ui_test/src/dependencies.rs

```diff
@@ -0,0 +1,64 @@
+use color_eyre::eyre::{bail, Result};
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use crate::Config;
+
+/// Compiles dependencies and returns the crate names and corresponding rmeta files.
+pub fn build_dependencies(config: &Config) -> Result<Vec<(String, PathBuf)>> {
+    let manifest_path = match &config.manifest_path {
+        Some(path) => path,
+        None => return Ok(vec![]),
+    };
+    let (program, args, envs): (&Path, &[_], &[_]) = match &config.dependency_builder {
+        Some(db) => (&db.program, &db.args, &db.envs),
+        None => (Path::new("cargo"), &[], &[]),
+    };
+    let mut build = Command::new(program);
+
+    // Avoid poisoning the sysroot and causing unnecessary rebuilds.
```

Comment:
sysroot? Why sysroot?

Reply:
that's just me getting confused by the "sysroot available at" message, which was also emitted, along with a rebuild of miri